File: Top 100 Codes/003-Sum of First N Natural numbers/Method-3-Using Recursion/Natural_Number.java
import java.util.Scanner;

public class Natural_Number {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter a Number:");
        int n = sc.nextInt();

        int sum = getsum(n);

           System.out.println(sum);
        
           sc.close();
        
    }

    static int getsum(int n)
    {
        if( n == 0)
          return 0;

        return n + getsum(n -1);
    }
    
}

File: Top 100 Codes/003-Sum of First N Natural numbers/Method-3-Using Recursion/NaturalNumber.cpp
#include<bits/stdc++.h>
using namespace std;

int getsum(int n)
{
    if(n == 0)
       return 0;

    return n + getsum(n - 1);
}

int main(){
    int n;
    cout << "Enter a Number : ";
    cin >> n;

    int sum = getsum(n);

    cout << sum;

    return 0;
}